analyzers/src/SonarAnalyzer.Common/Rules/UseUnixEpochBase.cs

```diff
@@ -57,23 +58,29 @@ public abstract class UseUnixEpochBase<TSyntaxKind, TLiteralExpression, TMemberA
                 Language.GeneratedCodeRecognizer,
                 c =>
                 {
-                    var literalsArguments = Language.Syntax.ArgumentExpressions(c.Node).OfType<TLiteralExpression>();
-                    if (!literalsArguments.Any(x => IsValueEqualTo(x, EpochYear) || literalsArguments.Count(x => IsValueEqualTo(x, EpochMonth)) != 2))
+                    var arguments = Language.Syntax.ArgumentExpressions(c.Node);
+                    var literalsArguments = arguments?.OfType<TLiteralExpression>();
+
+                    if (literalsArguments.Any(x => IsValueEqualTo(x, EpochYear)
```

Comment:
The two if statements can be merged if you move `CheckAndGetTypeName` to the end of the conditions.

Reply:
Unfortunately, the conditions are different: the first one is checking whether you have 3 literals (2 with number `1`, 1 with number `1970`) and then proceeds to check the type and constructor.
The second one is only searching for the constructor with a single argument of `621355968000000000` or pointing at a const with that value, and then proceeds to check semantically the type (not the constructor, since we have only 1 ctor with 1 argument for DateTime/DateTimeOffset).
Although it's feasible to merge everything in a single if statement, all the variants I tried are really cumbersome to read and understand. Happy to discuss it further